### ISSUE TITLE
iOS 11.3 HeaterCooler Service

### ISSUE THREAD

**Issue by nitaybz** [homebridge/HAP-NodeJS#577]
Hi!
I created the homebridge plugin [homebridge-tado-ac](https://github.com/nitaybz/homebridge-tado-ac) which rely on the HeaterCooler Service.

Since iOS 11.3 there has been a change in the way Siri communicate with the AC device... Maybe you can help with some of them through here... and maybe some are apple bugs:

1. Old issue - when asking Siri for the temperature in a room (where HeaterCooler exists), it gives you the lowest Threshold temperature and NOT the "Current Temperature", though I do see that the function called is the one for the current temperature, for example:
I ask: "What is the temperature in the Bedroom?", I get: "The Bedroom temperature AC is at 16°", and the logs show: "Current temperature is "23°".
Weird! isn't it?

2. New issue since 11.3 update - It's not possible to change the temperature.
I ask: "Set The Bedroom AC to 22°", I get: "Hmm, There are no devices in the Bedroom that support that action", thought power on an off are working properly.
a work around I found is to tell Siri to set a range between x to x and than she would set it to the higher one while replying that she is setting the range.

In conclusion, I believe this issue is coming from the fact that in this service there is no "Target Temperature" characteristic and only "CoolingThresholdTemperature" and "HeatingThresholdTemperature", which might confuse Siri...

I'm sure you recall I submitted an issue about it #500 , up until now I could manage without it but no more...

Is it something that can be fixed by you?

Thanks again for all your efforts !

**Comment by llemtt** [comment]
I haven't tried with Siri yet because my plugin is currently being developed... but I added the TargetTemperature characteristic (actually called getCharacteristic without even thinking it wasn't there) and it works.

**Comment by nitaybz** [comment]
I'm getting this error:
`HAP Warning: Characteristic 00000035-0000-1000-8000-0026BB765291 not in required or optional characteristics for service 000000BC-0000-1000-8000-0026BB765291. Adding anyway.`

and it's not really getting the target temperature or setting it when I change something in Home App
